tide_authenticated_content.install

```diff
@@ -47,6 +55,71 @@ function tide_authenticated_content_update_8002() {
  */
 function tide_authenticated_content_update_8003() {
   module_load_include('inc', 'tide_core', 'includes/helpers');
-  $config_location = [drupal_get_path('module', 'tide_authenticated_content') . '/config/install'];
+  $config_location = [
+    drupal_get_path('module',
+    'tide_authenticated_content') . '/config/install',
+  ];
   _tide_import_single_config('tide_authenticated_content.module', $config_location, TRUE);
 }
+
+/**
+ * Implements hook_uninstall().
+ */
+function tide_authenticated_content_uninstall() {
+  // Remove permissions on tide_authenticated_content to Approver and Editor.
+  $roles = ['approver', 'editor'];
+  $permissions = [
+    'create terms in authenticated_content',
+    'create terms in authenticated_content',
+  ];
+
+  foreach ($roles as $role_name) {
+    $role = Role::load($role_name);
+    if ($role) {
+      foreach ($permissions as $permission) {
+        $role->revokePermission($permission);
+      }
+      $role->save();
+    }
+  }
+  // Remove taxonomy and fields that tide_authenticated_content has installed.
+  Vocabulary::load('authenticated_content')->delete();
```

Comment:
Please verify there is content before to delete.

Reply:
same reason as above.